lit-flip.md

```diff
@@ -0,0 +1,323 @@
+
+Whichever side of my mind I might be in agreement with in any given moment, I created the [streaming listener directive](https://webcomponents.dev/edit/ar6PrCzLTsv9wunVnifw) to better support this reality. On top of maintaining the stateful progression of a stream, a streaming listener allows for binding fewer events at runtime by using the current state of the stream to determine what binding to do which can improve performance as well. Take a look at how this might be leveraged:
+
+```javascript
+import { streamingListener } from "./streaming-listener";
+
+// ...
+
+<input
+     type="range"
+     min="0"
+     max="100"
+     @manage=${streamingListener(
+         { type: "input", fn: this.start },
+         { type: "input", fn: this.stream },
+         { type: "change", fn: this.end }
+     )}
+/>
+```
+
+Here the directive supports the ability to bind `input` events to both `this.start` and `this.stream` depending on the state of the stream. This allows for only a single event to be bound to the `<input>` at any one time without you needing to manage this (or any other state in regards to your event listening) locally increasing performance and reducing the chances of copy/paste centric bugs when leveraged across multiple contexts.
+
+While I've made some feature additions and API changes when going between the [v1.0](https://webcomponents.dev/edit/ar6PrCzLTsv9wunVnifw?file=src%2Fstreaming-listener.ts) and [v2.0](https://webcomponents.dev/edit/collection/PfCT8IzVVjUxI3JiaF6x/w9bQYjy7dvlMEKW4MDJY?file=src%2Fstreaming-listener.ts) implementations, the biggest benefit of the class syntax that I see if the ability to more directly keep the state necessary to empower the directive. Previously this was done through the use of the following `WeakMap`s:
+
+```javascrip
+const previousValues = new WeakMap<
+  Part,
+  {
+    start: { type: string; fn: (event) => void };
+    stream: { type: string; fn: (event) => void };
+    end: { type: string; fn: (event) => void };
+    removeEventListeners: () => void;
+  }
+>();
+
+const stateMap = new WeakMap<Part, boolean>();
+```
+
+With these hanging around in the module scope, we are able to take advantage of the idea that the `Part` representing the location of the directive in the template is an object that keeps identity across multiple renders, which allows us to access to stored state on subsequent render passes. However, this can feel a little magic... why is this `Part` always the same? Can I really rely on that? Why did I make `previousValues` and `stateMap` separate? Oh, wait, that's not about magic, that's just me code reviewing myself...
```

Comment:
```suggestion
With these hanging around in the module scope, we are able to take advantage of the idea that the `Part` representing the location of the directive in the template is an object that keeps identity across multiple renders, which allows us access to stored state on subsequent render passes. However, this can feel a little magic... why is this `Part` always the same? Can I really rely on that? Why did I make `previousValues` and `stateMap` separate? Oh, wait, that's not about magic, that's just me code reviewing myself...
```